nrf_security/src/mbedtls/CMakeLists.txt

```diff
@@ -121,43 +121,43 @@ PREPEND(src_tls_replacement ${NRF_MBEDTLS_ROOT}/src/mbedtls/replacements/
 zephyr_include_directories(${ARM_MBEDTLS_ROOT}/include)
 zephyr_include_directories(${ARM_MBEDTLS_ROOT}/include/mbedtls)
 
-add_library(mbedcrypto_vanilla STATIC "")
-set(ZEPHYR_CURRENT_LIBRARY mbedcrypto_vanilla)
+add_library(${IMAGE}mbedcrypto_vanilla STATIC "")
```

Comment:
Could you use `zephyr_library_named()` here? To avoid having to set the current library afterwards.

Reply:
I remember at early days, there were some issues with that in relation to symbol renaming, but can't recall the exact nature of the problem, and thus, for this particular fix I consider it safest not to change too much.

But I agree, it is worth to investigate if this can be simplified, but I would suggest a separet commit and PR for that, to keep such changes independent of this PR.

Reply:
Followup, not using `zephyr_library_named()` allows the lib not to go into `whole-archive` and thus only being linked if needed, in same fashion as when Zephyr links to externally mbedtls library.

Reply:
Ok, but what is the reason for needing to not be whole-archived here? 

I'm asking since the rest of the zephyr libraries are created in that fashion, what is the difference here?

Reply:
will do.